### ISSUE TITLE
R&D: Collectors landscape page

### ISSUE THREAD

**Issue by cakrit** [netdata/netdata#5045]
##### Feature idea summary
Our collectors are shown in long lists and/or tables in the following places:
https://github.com/netdata/netdata/blob/master/README.md#what-does-it-monitor
https://github.com/netdata/netdata/blob/master/docs/Add-more-charts-to-netdata.md

We need to replace these views with an interactive collector's landscape, similar to 
https://github.com/cncf/landscape

The first version will have all the resources required to produce this view inside docs/generator. We can discuss making it better later.

I will try a spike to see what we can get quickly, so we can more properly estimate the required work later.

**Comment by paulfantom** [comment]
Maybe in the beginning something simpler would be better? Like this one: https://www.home-assistant.io/components/ ?

**Comment by cakrit** [comment]
Thanks, not quite as impressive, but I can definitely get images from it! 
Any additional suggestions are welcome. I'd really like to see something with the code that generates it publicly available, so that we don't have to build everything from scratch.

**Comment by cakrit** [comment]
It took be almost a whole day because I didn't know anything about node, yarn, npm etc, but I was able to do the following:
- Fork the two projects of the CNCF landscape, https://github.com/cncf/landscape and https://github.com/cncf/landscapeapp. landscape depends on the app and only contains configuration info (lists of apps and related artifacts). 
- Publish an npm package called `netdata-collectors-landscape` from my fork of the app.
- Make my landscape fork (let's just call it landscape) to read my npm package and build correctly
- Configure netlify to publish a separate site from my landscape project.

Main findings this far:
- There's no way that we'll be putting any of those artifacts and code into the netdata repository. We'll need to keep the two additional projects. 
- We can take advantage of an already built-in fetcher and adapt it to get info from netdata/netdata, the go plugin etc, to dynamically enrich information we'll have in the landscape project. I haven't looked at how exactly this works so I'm not sure how long it would take, but we can start with minimum integration and improve it later.
- The class and attribute names are hardcoded in several of the ~78 .js files of the app, so we'll need to visit them and adapt them to our needs (probably ~1 week effort to have a prototype and one more to incorporate feedback and have the final version). 
- The basic content is in a single file in the landscape, so it won't be too difficult to create a new one, with content from our main README and our existing directory structure + perhaps links to readmes. A couple of days will be more than enough to get it populated with basic data. The tricky part is mapping existing images, getting missing images etc. I will need to work on it a bit to see what will work. 

To sum up, I believe we can easily have a prototype in one sprint and a final version in a second sprint, with room for some other work. 
Assumption: little to no integration with the netdata/netdata project. 
Risk: The time required to gather and associate missing images

**Comment by cakrit** [comment]
@ktsaou agreed to proceed with the CNCF landscape app, so closing this spike as completed.
